geopyspark/geotrellis/render.py

```diff
@@ -74,8 +76,14 @@ def __init__(self, pyramid, ramp_name, debug=False):
         self.rdd_type = level0.rdd_type
         self.layer_metadata = list(map(lambda lev: lev.layer_metadata, pyramid))
         self.max_zoom = level0.zoom_level
-        self.pngpyramid = list(map(lambda layer: self.geopysc._jvm.geopyspark.geotrellis.PngRDD.asSingleband(layer.srdd, ramp_name), pyramid))
+        histogram = level0.get_histogram()
+        if level0.is_floating_point_layer():
+            mapper = lambda layer: self.geopysc._jvm.geopyspark.geotrellis.PngRDD.asSingleband(layer.srdd, histogram, ramp_name)
+        else:
+            mapper = lambda layer: self.geopysc._jvm.geopyspark.geotrellis.PngRDD.asIntSingleband(layer.srdd, histogram, ramp_name)
+        self.pngpyramid = list(map(mapper, pyramid))
```

Comment:
This might be less efficient than just doing list comprehension https://stackoverflow.com/a/1247490